pkg/sql/contention/txnidcache/fifo_cache.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package txnidcache
+
+import (
+	"sync"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/contention/contentionutils"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/uuid"
+	"github.com/cockroachdb/errors"
+)
+
+type fifoCache struct {
+	pool     *sync.Pool
+	capacity contentionutils.CapacityLimiter
+
+	mu struct {
+		syncutil.RWMutex
+
+		data map[uuid.UUID]roachpb.TransactionFingerprintID
+
+		eviction evictionList
+	}
+}
+
+type evictionNode struct {
```

Comment:
I think these things need to be pooled. I'd like to see the benchmark allocating on average zero or near zero bytes as opposed to the current ~70. I wouldn't focus so much on the runtime in terms of nanoseconds for this, if you get the allocations to zero, my money is that we won't see big impact on the kv workload.